shared/factory-generator/processor/src/jvmMain/kotlin/app/futured/factorygenerator/processor/PoetFactoryComponentGenerator.kt

```diff
@@ -0,0 +1,175 @@
+package app.futured.factorygenerator.processor
+
+import com.google.devtools.ksp.processing.CodeGenerator
+import com.google.devtools.ksp.symbol.KSClassDeclaration
+import com.google.devtools.ksp.symbol.KSValueParameter
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.ksp.toTypeName
+import com.squareup.kotlinpoet.ksp.writeTo
+
+/**
+ * `PoetFactoryComponentGenerator` is responsible for generating the factory object for a given component.
+ *
+ * This object utilizes KotlinPoet to generate the necessary code for creating a factory that
+ * uses Koin for dependency injection. It analyzes a component's constructor parameters to
+ * determine which dependencies need to be manually injected and generates a `createComponent` function
+ * that retrieves these dependencies from the Koin container.
+ *
+ * The generated factory object implements the `KoinComponent` interface and provides a
+ * `createComponent` function to instantiate the target component.
+ *
+ * Key functionalities:
+ * 1.  **Generates Factory Objects:** Creates a factory object for each component annotated
+ *     with a specific annotation (implicitly handled by the ksp processor).
+ * 2.  **Handles Dependency Injection:** Uses Koin's `get()` function to retrieve necessary
+ *     dependencies from the Koin container.
+ * 3.  **Supports Constructor Parameters:** Inspects the component's constructor to identify
+ *     parameters requiring manual injection (marked with `@InjectedParam`).
+ * 4.  **Handles Special Types:** Detects and handles special types like `AppComponentContext`,
+ *     `Navigation`, and `Arg` for specific component needs.
+ */
+object PoetFactoryComponentGenerator {
+
+    private const val INJECTED_PARAM_ANNOTATION = "InjectedParam"
+    private const val APP_COMPONENT_CONTEXT_TYPE_NAME = "AppComponentContext"
+    private const val NAVIGATION_TYPE_NAME = "Navigation"
+
+    private object Imports {
+        const val KOIN_COMPONENT_PACKAGE = "org.koin.core.component"
+        const val KOIN_PARAMETER_PACKAGE = "org.koin.core.parameter"
+        const val KOIN_COMPONENT_CLASS_NAME = "KoinComponent"
+        const val KOIN_GET_FUNCTION_NAME = "get"
+        const val KOIN_PARAMETERS_OF_FUNCTION_NAME = "parametersOf"
+    }
+
+    fun generateFactory(
+        factoryComponent: KSClassDeclaration,
+        codeGenerator: CodeGenerator,
+    ) {
+        // Component name without package e.g. FirstComponent
+        val baseName: String = factoryComponent.qualifiedName?.asString()?.substringAfterLast('.')
+            ?: error("Unable to get base name")
```

Comment:
If something goes wrong, I, as a user, would like to know why base name could not be generated

```suggestion
            ?: error("Unable to get base name, component qualified name: ${factoryComponent.qualifiedName?.asString()}")
```